corehq/form_processor/backends/sql/processor.py

```diff
@@ -243,23 +244,27 @@ def get_cases_from_forms(case_db, xforms):
         return touched_cases
 
     @staticmethod
-    def hard_rebuild_case(domain, case_id, detail):
-        try:
-            case = CaseAccessorSQL.get_case(case_id)
-            assert case.domain == domain
-            found = True
-        except CaseNotFound:
+    def hard_rebuild_case(domain, case_id, detail, lock=True):
+        case, lock_obj = FormProcessorSQL.get_case_with_lock(case_id, lock=lock)
+        if not case:
             case = CommCareCaseSQL(case_id=case_id, domain=domain)
-            found = False
+            if lock_obj:
+                lock_obj = CommCareCaseSQL.get_obj_lock_by_id(case_id)
```

Comment:
I'm not understanding this. Can the `lock_obj` returned by `FormProcessorSQL.get_case_with_lock` not be used here? Looks like at the very least it [should be released](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/dimagi/utils/couch/__init__.py#L180-L182) before being replaced.

Reply:
Oops, I refactored the name didn't review carefully enough.